### DISCUSSION TITLE
src/instrumentation.js not loading in production mode

### DISCUSSION THREAD

**Discussion by bigdefect** [vercel/next.js#68420] [Help]
### Summary

Hey folks, I'm fairly new to this node stack. In trying to set up [Sentry's Next.js integration](https://docs.sentry.io/platforms/javascript/guides/nextjs/), it worked fine in development (`next dev`), but not in production mode (`next start`).

```js
src/instrumentation.js
export async function register() {
  if (process.env.NEXT_RUNTIME === "nodejs") {
    await import("./sentry.server.config");
  }
  ...
}
```

If I move the file to the project root, `next start` does load it. Any ideas why?

### Additional information

https://nextjs.org/docs/app/api-reference/file-conventions/instrumentation

Based on this, I did confirm that the `instrumentationHook` experiment is indeed showing enabled.

```
> npx next build && npx next start
  ▲ Next.js 14.2.1
  - Environments: .env.local, .env
  - Experiments (use with caution):
    · instrumentationHook
...
```

**Comment by lforst** [comment]
In what way did it not work in production mode?

**Reply by bigdefect** [reply]
It wasn't being loaded, as evidenced by the startup log snippet I shared of it being loaded.

**Comment by nikolayemrikh** [comment]
`instrumentation.ts` placed in project root doesn't execute in 15.2.4 and 15.3.1 in production build. But executes if placed in src

**Reply by estevao-fay-deel** [reply]
What if you're not using a src directory in your project?

**Comment by Chimochikhin** [comment]
I can say more - instrumentation-client.ts - does not run in SRC folder.
Only if it inside root folder!

**Comment by ruisilva450** [comment]
I found out that `instrumentation[-client].ts` only runs on `next start` if they are placed on the same level as `pages` and `app`

```md
|                                       Root directory structure
├── src                                 <-- The source code for your service.
    └── pages                           <-- The source code for Next.js [pages](https://nextjs.org/docs/basic-features/pages) routing.
    └── instrumentation[-client].ts     <-- Instrumentation hooks for client-side and server-side.
```
```md
|                                   Root directory structure
├── src                             <-- The source code for your service.
|── pages                           <-- The source code for Next.js [pages](https://nextjs.org/docs/basic-features/pages) routing.
|── instrumentation[-client].ts     <-- Instrumentation hooks for client-side and server-side.
```

**Comment by robinkarlberg** [comment]
Incredible how it still works in dev but not in prod. And there is no clear message stating this if I'm not mistaken. 

But thanks for the solution!
